osf/models/registrations.py

```diff
@@ -567,10 +647,154 @@ def status_logs(self):
+        }
+
+    @property
+    def visible_contributors(self):
+        # Override for ContributorMixin
+        return OSFUser.objects.filter(
+            draftregistrationcontributor__draft_registration=self,
+            draftregistrationcontributor__visible=True
+        ).order_by(self.order_by_contributor_field)
+
+    @property
+    def contributor_email_template(self):
+        # Override for ContributorMixin
+        return 'draft_registration'
+
+    @property
+    def institutions_url(self):
+        # For NodeInstitutionsRelationshipSerializer
+        path = '/draft_registrations/{}/institutions/'.format(self._id)
+        return api_v2_url(path)
+
+    @property
+    def institutions_relationship_url(self):
+        # For NodeInstitutionsRelationshipSerializer
+        path = '/draft_registrations/{}/relationships/institutions/'.format(self._id)
+        return api_v2_url(path)
+
+    def update_search(self):
+        # Override for AffiliatedInstitutionMixin, not sending DraftRegs to search
+        pass
+
+    def can_view(self, auth):
+        """Does the user have permission to view the draft registration?
+        Checking permissions directly on the draft, not the node.
+        """
+        if not auth:
+            return False
+        return auth.user and self.has_permission(auth.user, READ)
+
+    def get_addons(self):
+        # Override for ContributorMixin, Draft Registrations don't have addons
+        return []
+
+    # Override Taggable
+    def add_tag_log(self, tag, auth):
+        self.add_log(
+            action=DraftRegistrationLog.TAG_ADDED,
+            params={
+                'draft_registration': self._id,
+                'tag': tag.name
+            },
+            auth=auth,
+            save=False
+        )
+
+    @property
+    def license(self):
+        if self.node_license_id:
+            return self.node_license
+        return None
+
+    @property
+    def all_tags(self):
+        """Return a queryset containing all of this draft's tags (incl. system tags)."""
+        # Tag's default manager only returns non-system tags, so we can't use self.tags
+        return Tag.all_tags.filter(draftregistration_tagged=self)
+
+    @property
+    def system_tags(self):
+        """The system tags associated with this draft registration. This currently returns a list of string
+        names for the tags, for compatibility with v1. Eventually, we can just return the
+        QuerySet.
+        """
+        return self.all_tags.filter(system=True).values_list('name', flat=True)
+
     @classmethod
-    def create_from_node(cls, node, user, schema, data=None, provider=None):
+    def create_from_node(cls, user, schema, node=None, data=None, provider=None):
         if not provider:
             provider = RegistrationProvider.load('osf')
+
+        if not node:
+            # If no node provided, a DraftNode is created for you
+            node = DraftNode.objects.create(creator=user, title='Untitled')
```

Comment:
Node is no longer required! If you don't pass one in, a `DraftNode` will be created for you. I gave it a default title, but should check with product.